### DISCUSSION TITLE
Mavlink-router over raspberry Zero 2W

### DISCUSSION THREAD

**Discussion by kickfliph** [mavlink-router/mavlink-router#319] [Q&A]
Hello there,

I'm trying to install Mavlink-router over a Raspberry Zero 2W. I'm following the installation instruction on this git webpage. It looks like it's going but it is taking too much time to compile, in this point It's stuck at "[2/15] Compiling C++ object src/mavlink-routerd.p/main.cpp.o", the rpiZw it's alive because it's reacting to my commands but the compiling process is not moving forward. I really appreciate if anyone can give your appreciation of this situation. 

Regards,

**Comment by jbeyerstedt** [comment]
The Raspberry Zero is really quite slow and will get even slower when running for a long time and the CPU temperature rises. So it'll either take quite some time or you even need to add some more cooling.
As discussed in https://github.com/mavlink-router/mavlink-router/issues/320, cross-compiling might be a solution to get a binary faster. But let's continue with the discussion in the issue.

**Comment by PeterJBurke** [comment]
I suggest you use the  -j 1 flag to ninja commands to limit the build to a single process. This should prevent the build from exceeding available memory on devices like the Raspberry Pi Zero and the Raspberry Pi 3.

See also the comment about it in another context here:
https://www.raspberrypi.com/documentation/computers/camera_software.html#udp
